CMakeLists.txt

```diff
@@ -34,6 +34,10 @@ include(pybind11)
 include(GoogleTest)
 include(GoogleBenchmark)
 
+if("${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}" VERSION_LESS_EQUAL "3.4")
```

Comment:
Would this be equivalent (and less code)?
```suggestion
if("${PYTHON_VERSION_STRING}" VERSION_LESS_EQUAL "3.4.0")
```

Reply:
Also should the check be less, rather than less than?

Reply:
If it works, happy to accept the changes.

Reply:
3.4.0 doesn't work. In fact I'm not aware that we have tested. below 3.6. Maybe the check should be Less 3.6 is failure?

Reply:
Actually this doesn't work. Not sure what is setting those variables but it does not seem to be [`FindPython`](https://cmake.org/cmake/help/v3.15/module/FindPython.html#module:FindPython).

The version requirements here (> 3.4) are different to the changes in the readme (>= 3.4).

Reply:
It's PyBind11 not `FindPython`

Reply:
A part of me had hoped pybind11 would have just used FindPython...

(they do use FindPythonInterp, which should still set `PYTHON_VERSION_STRING`)

Reply:
Great. I'll make the changes you've proposed.

Reply:
Actually, don't bother. `PYTHON_VERSION_STRING` is still not set...

Reply:
Unfortunately this Cmake var is not propagated through, shame as it would have been a neater solution.